### PR TITLE
Add ModalPresentationConfiguration for transition and presentation style

### LINKED ISSUE #2: Presentation- & transition-style in ModalNavigationPresenter
For compatibility reason, there is a need to change the default modal presentation to the old style. It could be much handy if custom styles for presentation/transition can be applied before the begin of a transition.

### PR DESCRIPTION
**Pull request by SimonNumberTwo**
# Description
This MR adds the possibility to directly give a configuration for the `ModalPresenter` and the `ModalNavigationPresenter` to set the `modalTransitionStyle` as well as the `modalPresentationStyle`

# Link to issue
Closes #2

### CODE REVIEW

**Review by JensK611 — COMMENTED** [review]
This is looking good! It would be great if you can add some documentation in the readme. Just one or two sentences are enough.

**Comment by SimonNumberTwo** [comment]
Added a simple note in the README about the newly added configuration possibilities

File: JLCoordinator/Sources/Presenter/PresenterImplementations/ModalNavigationPresenter.swift [hunk 1]
```diff
@@ -69,6 +79,10 @@ public class ModalNavigationPresenter: ModalPresenting, NavigablePresenting {
 }
 
 extension ModalNavigationPresenter: AdaptivePresentationControllerDelegate {
+    func adaptivePresentationStyle(for controller: UIPresentationController) -> UIModalPresentationStyle {
+        return modalPresentationConfiguration.presentationStyle
```
Comment (jjorn) [inline JLCoordinator/Sources/Presenter/PresenterImplementations/ModalNavigationPresenter.swift:82]: `Return` can be omitted

File: JLCoordinator/Sources/Presenter/PresenterImplementations/ModalPresenter.swift [hunk 2]
```diff
@@ -5,18 +5,25 @@ open class ModalPresenter: ModalPresenting {
 
     public let presentingViewController: UIViewController
     private let adaptivePresentationDelegateWrapper: AdaptivePresentationControllerDelegateWrapper = .init()
+    private let modalPresentationConfiguration: ModalPresentationConfiguration
 
-    public init(presentingViewController: UIViewController) {
+    public init(
+        presentingViewController: UIViewController,
+        modalPresentationConfiguration: ModalPresentationConfiguration = .default
```
Comment (jjorn) [inline JLCoordinator/Sources/Presenter/PresenterImplementations/ModalPresenter.swift]: (Personally) I think `modalPresentationConfiguration:` is too long. Can we keep this short?

**Review by jjorn — COMMENTED** [review]
Could you please add sample code in example?

I've try to implement the changes in example and got following error:

```
2021-04-26 09:09:19.236399+0200 JLCoordinatorExample[21505:731750] [Assert] Setting modalPresentationStyle once presentationController has been accessed will have no effect until <UINavigationController: 0x7fdec481bc00> is presented, dismissed, and presented again.
```

code in ViewCoordinator.swift: 

```
func didTriggerModalNavigationController(in viewController: ViewController) {
        let modalConfig = ModalPresentationConfiguration(
            transitionStyle: .crossDissolve,
            presentationStyle: .fullScreen
        )
        let viewCoordinator: ViewCoordinator = .init(
            presenter: ModalNavigationPresenter(presentingViewController: viewController, modalPresentationConfiguration: modalConfig)
        )
        add(childCoordinator: viewCoordinator)
        viewCoordinator.start()
    }
```

File: JLCoordinator/Sources/Presenter/PresenterImplementations/ModalNavigationPresenter.swift (hunk 1 above)
Comment (SimonNumberTwo) [reply JLCoordinator/Sources/Presenter/PresenterImplementations/ModalNavigationPresenter.swift:82]: I would like to keep the return statement to be more explicit.

File: JLCoordinator/Sources/Presenter/PresenterImplementations/ModalPresenter.swift (hunk 2 above)
Comment (SimonNumberTwo) [reply JLCoordinator/Sources/Presenter/PresenterImplementations/ModalPresenter.swift]: I renamed it to `configuration` in the initialiser of both modal presenter.

**Review by jjorn — APPROVED** [review]

**Pull request merged**